PyU4V/tools/openstack/migrate_utils.py

```diff
@@ -0,0 +1,866 @@
+                self.print_pretty_table(message)
+            storagegroup = self.get_storage_group(
+                element_dict['new_sg_name'])
+        else:
+            storagegroup = self.get_or_create_cascaded_storage_group(
+                element_dict)
+        if storagegroup:
+            port_group = element_dict['port_group']
+            initiator_group = element_dict['initiator_group']
+            self.conn.provisioning.create_masking_view_existing_components(
+                port_group, element_dict['new_mv_name'],
+                storagegroup['storageGroupId'],
+                host_name=initiator_group)
+        else:
+            exception_message = (
+                'Cannot create or find the storagegroup.')
+            self.smart_print(exception_message, ERROR)
+            raise exception.ResourceNotFoundException(
+                data=exception_message)
+
+    def get_masking_view(self, mv_name):
+        """Get the masking view object from the name.
+
+        :param mv_name: masking view name
+        :returns: dict
+        """
+        masking_view = None
+        try:
+            masking_view = self.conn.provisioning.get_masking_view(
+                mv_name)
+        except exception.ResourceNotFoundException:
+            print_str = 'Masking view %s not found.'
+            self.smart_print(print_str, WARNING, mv_name)
+        return masking_view
+
+    def get_or_create_masking_view(
+            self, element_dict, portgroup, host, revert=False):
+        """Get or create masking view from component elements.
+
+        :param element_dict: element dictionary
+        :param portgroup: port group name
+        :param host: host name
+        :param test: is it a test case
+        :returns: dict
+        """
+        new_masking_view_details = self.get_masking_view(
+            element_dict['new_mv_name'])
+        if new_masking_view_details:
+            if self.validate_existing_masking_view(
+                    new_masking_view_details, portgroup, host,
+                    element_dict, revert):
+                print_str = (
+                    'The existing masking view %s will be used.')
+                self.smart_print(
+                    print_str, DEBUG, element_dict['new_mv_name'])
+                self.smart_print('\n', DEBUG)
+            else:
+                print_str = (
+                    'Something is wrong with the existing masking view %s.')
+                self.smart_print(
+                    print_str, WARNING, element_dict['new_mv_name'])
+        else:
+            print_str = 'Creating masking view %s.'
+            self.smart_print(print_str, DEBUG, element_dict['new_mv_name'])
+            self.get_or_create_elements(element_dict, revert)
+            new_masking_view_details = self.get_masking_view(
+                element_dict['new_mv_name'])
+        return new_masking_view_details
+
+    def move_vols_from_source_to_target(
```

Comment:
vols is fine here but rest of methods use volumes if you want to keep it consistent

Reply:
sure